examples/v3/multipart/consumer.spec.ts

```diff
@@ -0,0 +1,231 @@
+          body: form, // The multipart form data as a string
+          headers, // Headers including Content-Type with boundary
+        },
+        willRespondWith: {
+          status: 201,
+          body: {
+            // Use a matcher to flexibly match any string value for 'foo'
+            foo: MatchersV3.like('bar'),
+          },
+        },
+      })
+      // Execute the test against the mock server
+      .executeTest(async (mockServer) => {
+        // Send the actual request to the mock server with the form data
+        const response = await axios.post(`${mockServer.url}/test`, form, {
+          headers,
+        });
+
+        // Verify the response contains the expected 'foo' field with value 'bar'
+        expect(response.data.foo).to.equal('bar');
+      });
+  });
+
+  it('creates a pact to verify with multipart file upload', async () => {
+    // Create a temporary file to upload
+    const fs = require('fs');
+    const path = require('path');
+    const tmpFile = path.join(__dirname, 'test-file.txt');
+    fs.writeFileSync(tmpFile, 'This is test file content');
+
+    // Create form data with a file upload
+    // Read the file content synchronously and append to FormData
+    const formData = new FormData();
+    const fileContent = fs.readFileSync(tmpFile);
+    formData.append('document', fileContent, 'test-file.txt');
+
+    // Extract the headers that include the Content-Type with the boundary parameter
+    const headers = formData.getHeaders();
+
+    // Convert the form data to a string representation for the request body
+    const form = formData.getBuffer().toString();
+
+    await pact
+      .addInteraction({
+        states: [{ description: 'a file upload is requested' }],
+        uponReceiving: 'a multipart file upload request',
+        withRequest: {
+          method: 'POST',
+          path: '/upload',
+          body: form, // The multipart form data as a string (including file content)
+          headers, // Headers including Content-Type with boundary
+        },
+        willRespondWith: {
+          status: 200,
+          body: {
+            // Use matchers to flexibly match any values in the response
+            success: MatchersV3.boolean(true),
+            message: MatchersV3.like('File uploaded successfully'),
+          },
+        },
+      })
+      // Execute the test against the mock server
+      .executeTest(async (mockServer) => {
+        // Send the actual multipart file upload request
+        const response = await axios.post(`${mockServer.url}/upload`, form, {
+          headers,
+        });
+
+        // Verify the response indicates success
+        expect(response.data.success).to.equal(true);
+        expect(response.data.message).to.equal('File uploaded successfully');
+
+        // Clean up the temporary file
+        fs.unlinkSync(tmpFile);
+      });
+  });
+
+  it('creates a pact using withRequestMultipartFileUpload helper method', async () => {
```

Comment:
Why do we have both ways of doing this? I might be missing the context, but this example should probably explain this in some detail.

Reply:
i was showing multiple ways of how one can pass multipart form data, either directly with normal fields or file fields, or built in methods like withRequestMultipartFileUpload with and without boundaries.

Reply:
ive added more details in the readme